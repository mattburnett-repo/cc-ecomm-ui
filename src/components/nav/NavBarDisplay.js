
import { Link } from 'react-router-dom'

import ShowCartButton from '../../widgets/ShowCartButton'
import GoHomeButton from '../../widgets/GoHomeButton'
import GoToCheckoutButton from '../../widgets/GoToCheckoutButton'
import LogoutButton from '../../widgets/LogoutButton'

import { StyledNavBar } from '../styles/NavBar.style'

export default function NavBarDisplay (props) {
    // const { handleLogoutClick } = props.handlers
    const { calledFrom, headerMessage, currentCartCount } = props

    return (
        <StyledNavBar>
            <div role="presentation" aria-label='nav-bar-placeholder' />
            <div role="presentation" aria-label="nav-bar-message">
                <div role="presentation" id="header-message" aria-label="header-message">{headerMessage}</div>
                {(calledFrom === 'dashboard') ? (
                    <>
                        <div role="presentation" aria-label="header-current-cart-item-count">
                            {(currentCartCount === 0) ?
                                (<div>You have {currentCartCount} items in your current cart</div>)
                                : ( <Link to ="/cart-detail"  aria-label="go-to-current-cart-link">You have {currentCartCount} item/s in your current cart</Link>)
                            }
                        </div>
                    </>
                ) : (<></>)}
            </div>
            <div role="presentation" aria-label="nav-bar-buttons">
                {(calledFrom === 'productDetail') ? (
                    <>
                        <ShowCartButton />
                        <GoHomeButton />
                        {/* <button aria-label="logout" onClick={handleLogoutClick}>Log Out</button> */}
                        <LogoutButton />
                    </>
                ) : (<></>)}
                {(calledFrom === 'cart') ? (
                    <>
                        <GoToCheckoutButton />
                        <GoHomeButton />
                        {/* <button aria-label="logout" onClick={handleLogoutClick}>Log Out</button> */}
                        <LogoutButton />
                    </>
                ) : (<></>)}
                {(calledFrom === 'shippingInfo' || calledFrom === 'paymentInfo' || calledFrom === "dashboard") ? (
                    <>
                        <GoHomeButton />  
                        {/* <button aria-label="logout" onClick={handleLogoutClick}>Log Out</button>  */}
                        <LogoutButton />
                    </>
                ) : (<></>)}
            </div>
        </StyledNavBar>
    )
}
