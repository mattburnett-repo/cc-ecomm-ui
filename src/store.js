import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;