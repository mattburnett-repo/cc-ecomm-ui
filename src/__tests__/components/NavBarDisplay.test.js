// 20211217: any wireframe will do...

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux'
import store from '../../store'
import { MemoryRouter as Router } from 'react-router-dom';

import NavBarDisplay from '../../components/nav/NavBarDisplay';

// FIXME: need a better way to keep track of cart_items count. this is an inadequate stub
// function mockCurrentCart() { 
//     return 0 // we just want to mock the length of the (empty) cart_items array
// }

function mockHandlers() {
    return {
        handleHomeClick: jest.fn(),
        handleShowCartClick: jest.fn(),
        handleCheckoutClick: jest.fn(),
        handleCancelClick: jest.fn(),
        handleLogoutClick: jest.fn()
    }
}

describe('<NavBarDisplay /> component tests', () => {
    it('should display a header message', () => { // TODO: check other components that have header messages, and refactor to here
        render(<Router><NavBarDisplay calledFrom="" handlers={mockHandlers()} /></Router>);
        screen.getByRole('presentation', {name: /header-message/i});
        // TODO: test conditional header messages in feature tests
    });
    it('should render Current Cart item count', () => {
        render(<Router><NavBarDisplay calledFrom="" handlers={mockHandlers()} /></Router>);
        screen.getByRole('presentation', { name: /header-current-cart-item-count/i})
    })
    it('renders and clicks a logout button', () => {
        render(<Provider store={store}><Router>
                    <NavBarDisplay calledFrom="dashboard" handlers={mockHandlers()} />
                </Router></Provider>);

        const theVal = screen.getByRole('button', { name: /logout/i});
        fireEvent.click(theVal);
    });

    it('renders and clicks a home button if user is not on dashboard', () => {
        render(<Provider store={store}><Router>
                    <NavBarDisplay calledFrom="login" handlers={mockHandlers()} />
                </Router></ Provider>);

        const theVal = screen.getByRole('button', { name: /home/i});
        fireEvent.click(theVal);
    });
    it('renders and clicks a show cart button if user is not on dashboard', () => {
        render(<Provider store={store}><Router>
                    <NavBarDisplay calledFrom="login" handlers={mockHandlers()} />
                </Router></Provider>);

        const theVal = screen.getByRole('button', { name: /show-cart/i});
        fireEvent.click(theVal);
    });
    it('renders and clicks a checkout button if user is on the cart page', () => {
        render(<Provider store={store}><Router>
                    <NavBarDisplay calledFrom="cart"  handlers={mockHandlers()} />
                </Router></Provider>);

        const theVal = screen.getByRole('button', { name: /checkout/i});
        fireEvent.click(theVal);
    });
    it('renders and clicks a cancel button if user is on the shipping info page', () => {
        render(<Provider store={store}><Router>
                    <NavBarDisplay calledFrom="shippingInfo" handlers={mockHandlers()} />
                </Router></Provider>);

        const theVal = screen.getByRole('button', { name: /cancel/i});
        fireEvent.click(theVal);
    });
    it('renders and clicks a cancel button if user is on the payment info page', () => {
        render(<Provider store={store}><Router>
                    <NavBarDisplay calledFrom="paymentInfo" handlers={mockHandlers()} />
                </Router></Provider>);

        const theVal = screen.getByRole('button', { name: /cancel/i});
        fireEvent.click(theVal);
    });

    it('renders a snapshot', () => {
        const tree = renderer.create(<Router><NavBarDisplay handlers={mockHandlers()}  /></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });
}); // end component