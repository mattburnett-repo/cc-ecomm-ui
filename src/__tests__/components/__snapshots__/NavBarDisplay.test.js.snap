// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarDisplay /> component tests renders a snapshot 1`] = `
Array [
  <div
    className="sc-bdvvtL gUxESt"
  >
    <div
      aria-label="nav-bar-container"
      className="nav-bar-container"
      role="presentation"
    >
      <div />
      <div
        aria-label="nav-bar-message"
        role="presentation"
      >
        <div
          aria-label="header-message"
          className="header-message"
          role="presentation"
        />
      </div>
      <div
        aria-label="nav-bar-buttons"
        role="presentation"
      />
    </div>
  </div>,
  <div
    className="sc-bdvvtL gUxESt"
  >
    <div
      aria-label="nav-bar-container"
      className="nav-bar-container"
      role="presentation"
    >
      <div />
      <div
        aria-label="nav-bar-message"
        role="presentation"
      >
        <div
          aria-label="header-message"
          className="header-message"
          role="presentation"
        />
      </div>
      <div
        aria-label="nav-bar-buttons"
        role="presentation"
      />
    </div>
  </div>,
]
`;
